### DISCUSSION TITLE
terraform destroy actions

### DISCUSSION THREAD

**Discussion by Alwinl69** [community/community#65686] [Actions]
### Select Topic Area

General

### Body

Hi , 

Can anyone say how to configure terraform destroy actions?

**Comment by orazmyradov** [comment]
@Alwinl69 
Terraform provides a `destroy` command to remove infrastructure resources that were previously provisioned. The `destroy` command uses the configuration files to determine which resources it needs to destroy. By default, it destroys all the resources specified in your Terraform configuration.

However, if you want to be more specific or configure the destruction process, here are some ways to do so:

1. **Targeted Destroy**:
   Use the `-target` argument to specify a particular resource to destroy:
   ```bash
   terraform destroy -target=aws_instance.my_instance
   ```
   You can use this option multiple times in one command to target multiple resources. This is useful if you don't want to destroy everything, just specific resources.

2. **Prevent Destruction of a Resource**:
   If you never want a particular resource to be destroyed, you can set the `prevent_destroy` lifecycle attribute for that resource in your Terraform configuration:
   ```hcl
   resource "aws_instance" "my_instance" {
     # ... other configuration ...

     lifecycle {
       prevent_destroy = true
     }
   }
   ```
   If you run `terraform destroy` with the above configuration, Terraform will return an error and won't destroy the resource. If you really need to destroy it, you'll have to remove or change the `prevent_destroy` attribute.

3. **Auto-approve**:
   By default, `terraform destroy` will show you a plan of what will be destroyed and ask for confirmation. If you're automating this or you're sure about the destruction, you can use the `-auto-approve` flag:
   ```bash
   terraform destroy -auto-approve
   ```
   This will bypass the confirmation prompt.

4. **Using a Different State File**:
   If you maintain different state files for different environments (e.g., prod, staging), you can specify which state file to use with the `-state` flag:
   ```bash
   terraform destroy -state=path/to/statefile.tfstate
   ```

5. **Using Variables**:
   If your Terraform configurations use variables, and you want to pass them during the destruction process, use the `-var` flag:
   ```bash
   terraform destroy -var="instance_type=t2.micro"
   ```

6. **Destroy with a Different Backend**:
   If you use remote backends to store your Terraform state, you can specify a different backend configuration during destruction using the `-backend-config` option.

Always be cautious when using `terraform destroy`, especially in production environments. It's a powerful command that can remove your entire infrastructure, leading to data loss or service outages. Always review the plan carefully before confirming the destruction.
